llvm/lib/Target/AIE/AIEPostSelectOptimize.cpp

```diff
@@ -303,6 +431,15 @@ bool AIEPostSelectOptimize::runOnMachineFunction(MachineFunction &MF) {
     Changed |= removeRedundantCopies(MBB, MF.getRegInfo());
   }
 
+  // 3. Duplicate addressing registers.
+  // As we use getPseudoMoveOpcode from TII, that is only available
+  // in AIE2.
+  if (!MF.getTarget().getTargetTriple().isAIE1()) {
```

Comment:
Nit: Maybe add `assert(TII->getPseudoMoveOpcode() && "Target must have a PseudoMove instruction");` inside the `if`. This way there will be a clear error message for future AIE versions.